diff --git a/jnius/env.py b/jnius/env.py
index ea12d82..4111845 100644
--- a/jnius/env.py
+++ b/jnius/env.py
@@ -288,7 +288,7 @@ class MacOsXJavaLocation(UnixJavaLocation):
 
 class AndroidJavaLocation(UnixJavaLocation):
     def get_libraries(self):
-        return ['SDL2', 'log']
+        return ['log', 'pyjni']
 
     def get_include_dirs(self):
         # When cross-compiling for Android, we should not use the include dirs
diff --git a/jnius/jnius_jvm_android.pxi b/jnius/jnius_jvm_android.pxi
index dae6b31..d1e27a4 100644
--- a/jnius/jnius_jvm_android.pxi
+++ b/jnius/jnius_jvm_android.pxi
@@ -1,6 +1,6 @@
 # on android, rely on SDL to get the JNI env
-cdef extern JNIEnv *SDL_AndroidGetJNIEnv()
+cdef extern JNIEnv *WebView_AndroidGetJNIEnv()
 
 
 cdef JNIEnv *get_platform_jnienv() except NULL:
-    return <JNIEnv*>SDL_AndroidGetJNIEnv()
+    return <JNIEnv*>WebView_AndroidGetJNIEnv()
diff --git a/setup.py b/setup.py
index 3cca8c1..a7f40f0 100644
--- a/setup.py
+++ b/setup.py
@@ -54,14 +54,7 @@ PXI_FILES = [
 EXTRA_LINK_ARGS = []
 
 # detect Python for android
-PLATFORM = sys.platform
-NDKPLATFORM = getenv('NDKPLATFORM')
-if NDKPLATFORM is not None and getenv('LIBLINK'):
-    PLATFORM = 'android'
-
-# detect platform
-if PLATFORM == 'android':
-    PYX_FILES = [fn[:-3] + 'c' for fn in PYX_FILES]
+PLATFORM = 'android'
 
 JAVA=get_java_setup(PLATFORM)
 
