diff --git a/rust/src/helpers.rs b/rust/src/helpers.rs
index 364075a..8c5fbe8 100644
--- a/rust/src/helpers.rs
+++ b/rust/src/helpers.rs
@@ -56,7 +56,7 @@ pub fn local_time(
         seconds -= (10957 * SECS_PER_DAY as usize) as isize;
         year += 30; // == 2000
     } else {
-        seconds += ((146_097 - 10957) * SECS_PER_DAY as usize) as isize;
+        seconds += ((146_097 - 10957) as u64 * SECS_PER_DAY as u64) as isize;
         year -= 370; // == 1600
     }
