diff --git a/setup.py b/setup.py
--- a/setup.py
+++ b/setup.py
@@ -54,6 +54,8 @@
     include_dirs = ['/home/even/gdal/3.10/build/port', '/home/even/gdal/3.10/port', '/home/even/gdal/3.10/build/gcore', '/home/even/gdal/3.10/gcore', '/home/even/gdal/3.10/alg', '/home/even/gdal/3.10/ogr/', '/home/even/gdal/3.10/ogr/ogrsf_frmts', '/home/even/gdal/3.10/gnm', '/home/even/gdal/3.10/apps']
     library_dirs = ['/home/even/gdal/3.10/build']
 libraries = ['gdal']
+if 'GDAL_LIBS' in os.environ:
+    libraries = os.environ['GDAL_LIBS'].split(',')


 # ---------------------------------------------------------------------------
@@ -228,6 +230,9 @@

     def get_gdal_config(self, option):
         try:
+            var_name = f"GDAL_{option.upper()}"
+            if var_name in os.environ:
+                return os.environ[var_name]
             return fetch_config(option, gdal_config=self.gdal_config)
         except gdal_config_error:
             msg = 'Could not find gdal-config. Make sure you have installed the GDAL native library and development headers.'
