diff --git a/setup.py b/setup.py
--- a/setup.py
+++ b/setup.py
@@ -55,6 +55,8 @@
     include_dirs = ['/home/runner/work/gdal/gdal/python_pkgs/port', '/home/runner/work/gdal/gdal/port', '/home/runner/work/gdal/gdal/python_pkgs/gcore', '/home/runner/work/gdal/gdal/gcore',  '/home/runner/work/gdal/gdal/gcore/multidim', '/home/runner/work/gdal/gdal/alg', '/home/runner/work/gdal/gdal/ogr/', '/home/runner/work/gdal/gdal/ogr/ogrsf_frmts', '/home/runner/work/gdal/gdal/gnm', '/home/runner/work/gdal/gdal/apps']
     library_dirs = ['/home/runner/work/gdal/gdal/python_pkgs']
 libraries = ['gdal']
+if 'GDAL_LIBS' in os.environ:
+    libraries = os.environ['GDAL_LIBS'].split(',')


 # ---------------------------------------------------------------------------
@@ -240,6 +242,9 @@

     def get_gdal_config(self, option):
         try:
+            var_name = f"GDAL_{option.upper()}"
+            if var_name in os.environ:
+                return os.environ[var_name]
             return fetch_config(option, gdal_config=self.gdal_config)
         except gdal_config_error:
             msg = 'Could not find gdal-config. Make sure you have installed the GDAL native library and development headers.'
