diff --git a/libxml2.syms b/libxml2.syms
index 370dcf1..3c4a3dc 100644
--- a/libxml2.syms
+++ b/libxml2.syms
@@ -453,16 +453,16 @@ LIBXML2_2.4.30 {
   xmlNanoFTPUpdateURL;
 
 # DOCBparser
-  docbCreateFileParserCtxt;
-  docbCreatePushParserCtxt;
-  docbEncodeEntities;
-  docbFreeParserCtxt;
-  docbParseChunk;
-  docbParseDoc;
-  docbParseDocument;
-  docbParseFile;
-  docbSAXParseDoc;
-  docbSAXParseFile;
+# docbCreateFileParserCtxt;
+# docbCreatePushParserCtxt;
+# docbEncodeEntities;
+# docbFreeParserCtxt;
+# docbParseChunk;
+# docbParseDoc;
+# docbParseDocument;
+# docbParseFile;
+# docbSAXParseDoc;
+# docbSAXParseFile;
 
 # xpath
   xmlXPathCastBooleanToNumber;
@@ -2187,7 +2187,7 @@ LIBXML2_2.6.29 {
     global:
 
 # threads
-  xmlDllMain;
+  # xmlDllMain;
 } LIBXML2_2.6.28;
 
 LIBXML2_2.6.32 {
