diff --git a/vendored-meson/meson/mesonbuild/linkers/linkers.py b/vendored-meson/meson/mesonbuild/linkers/linkers.py
index 7b3202c..2bf5e19 100644
--- a/vendored-meson/meson/mesonbuild/linkers/linkers.py
+++ b/vendored-meson/meson/mesonbuild/linkers/linkers.py
@@ -767,7 +767,7 @@ def get_allow_undefined_args(self) -> T.List[str]:
         return self._apply_prefix('-undefined,dynamic_lookup')
 
     def get_std_shared_module_args(self, options: 'KeyedOptionDictType') -> T.List[str]:
-        return ['-bundle'] + self._apply_prefix('-undefined,dynamic_lookup')
+        return ['-dynamiclib'] + self._apply_prefix('-undefined,dynamic_lookup')
 
     def get_pie_args(self) -> T.List[str]:
         return []
