diff --git a/setupinfo.py b/setupinfo.py
index 97e3399..634183a 100644
--- a/setupinfo.py
+++ b/setupinfo.py
@@ -1,12 +1,13 @@
-import sys
 import io
 import os
 import os.path
 import subprocess
-
-from setuptools.command.build_ext import build_ext as _build_ext
+import sys
 from distutils.core import Extension
 from distutils.errors import CompileError, DistutilsOptionError
+
+from setuptools.command.build_ext import build_ext as _build_ext
+
 from versioninfo import get_base_dir
 
 try:
@@ -330,7 +331,7 @@ def include_dirs(static_include_dirs):
     result = []
     possible_include_dirs = flags('cflags')
     for possible_include_dir in possible_include_dirs:
-        if possible_include_dir.startswith('-I'):
+        if possible_include_dir.startswith('-I') and not possible_include_dir.endswith('/MacOSX.sdk/usr/include'):
             result.append(possible_include_dir[2:])
     return result
 
