index 6ae2502..ab90b57 100755
--- a/builds/unix/config.sub
+++ b/builds/unix/config.sub
@@ -146,6 +146,7 @@ case $1 in
                        | uclinux-gnu* | kfreebsd*-gnu* | knetbsd*-gnu* | netbsd*-gnu* \
                        | netbsd*-eabi* | kopensolaris*-gnu* | cloudabi*-eabi* \
                        | storm-chaos* | os2-emx* | rtmk-nova* | managarm-* \
+                       | ios*-simulator | tvos*-simulator | watchos*-simulator \
                        | windows-* )
                                basic_machine=$field1
                                basic_os=$maybe_os
@@ -1492,6 +1493,8 @@ case $os in
                        ;;
                esac
                ;;
+       ios | ios-simulator)
+               ;;
        *)
                # No normalization, but not necessarily accepted, that comes below.
                ;;
@@ -1797,6 +1800,8 @@ case $kernel-$os in
                # None (no kernel, i.e. freestanding / bare metal),
                # can be paired with an output format "OS"
                ;;
+       ios-simulator)
+               ;;
        -*)
                # Blank kernel with real OS is always fine.
                ;;