diff --git a/fiona/__init__.py b/fiona/__init__.py
index 2971192..ab51dc5 100644
--- a/fiona/__init__.py
+++ b/fiona/__init__.py
@@ -28,6 +28,12 @@ from pathlib import Path
 import platform
 import warnings
 
+# Flet
+import os
+from os.path import dirname
+os.environ["GDAL_DATA"] = f"{dirname(dirname(__file__))}/opt/share/gdal"
+os.environ["PROJ_DATA"] = f"{dirname(dirname(__file__))}/opt/share/proj"
+
 if platform.system() == "Windows":
     _whl_dir = os.path.join(os.path.dirname(__file__), ".libs")
     if os.path.exists(_whl_dir):
diff --git a/setup.py b/setup.py
index e1f48a6..6c04d39 100644
--- a/setup.py
+++ b/setup.py
@@ -9,7 +9,6 @@ import sys
 from setuptools import setup
 from setuptools.extension import Extension
 
-
 # Ensure minimum version of Python is running
 if sys.version_info[0:2] < (3, 6):
     raise RuntimeError('Fiona requires Python>=3.6')
@@ -85,6 +84,13 @@ if 'clean' not in sys.argv:
         else:
             logging.warning("Failed to get options via gdal-config: %s", str(e))
 
+    if 'GDAL_LIB_PATH' in os.environ:
+        library_dirs.extend(os.environ['GDAL_LIB_PATH'].split(":"))
+    if 'GDAL_INCLUDE_PATH' in os.environ:
+        include_dirs.extend(os.environ['GDAL_INCLUDE_PATH'].split(":"))
+    if 'GDAL_LIBS' in os.environ:
+        libraries.extend(os.environ['GDAL_LIBS'].split(","))
+
     # Get GDAL API version from environment variable.
     if 'GDAL_VERSION' in os.environ:
         gdalversion = os.environ['GDAL_VERSION']
