diff --git a/blis/_src/frame/thread/bli_pthread.c b/blis/_src/frame/thread/bli_pthread.c
--- a/blis/_src/frame/thread/bli_pthread.c
+++ b/blis/_src/frame/thread/bli_pthread.c
@@ -319,7 +319,7 @@

 // -- pthread_barrier_*() --

-#if defined(__APPLE__) || defined(_MSC_VER)
+#if defined(__APPLE__) || defined(_MSC_VER) || defined(__ANDROID__)

 // For OS X and Windows, we define barriers ourselves in terms of the rest
 // of the API, though for slightly different reasons: For Windows, we must
diff --git a/blis/_src/include/linux-generic/blis.h b/blis/_src/include/linux-generic/blis.h
--- a/blis/_src/include/linux-generic/blis.h
+++ b/blis/_src/include/linux-generic/blis.h
@@ -1537,7 +1537,7 @@ typedef pthread_cond_t         bli_pthread_cond_t;
 typedef pthread_condattr_t     bli_pthread_condattr_t;
 typedef pthread_once_t         bli_pthread_once_t;

-#if defined(__APPLE__)
+#if defined(__APPLE__) || defined(__ANDROID__)

 // For OS X, we must define the barrier types ourselves since Apple does
 // not implement barriers in their variant of pthreads.
diff --git a/setup.py b/setup.py
--- a/setup.py
+++ b/setup.py
@@ -21,8 +21,13 @@
 import sys
 import platform
 import numpy
+import sysconfig

 MOD_NAMES = ["blis.cy", "blis.py"]
+
+# Redirect the compiler to CC
+os.environ["BLIS_ARCH"] = "generic"
+os.environ["BLIS_COMPILER"] = os.environ["CC"]

 print("BLIS_COMPILER?", os.environ.get("BLIS_COMPILER", "None"))

@@ -199,6 +204,9 @@
         objects = []
         platform_arch = platform + "-" + py_arch
         compiler = self.get_compiler_name()
+        host_triplet = sysconfig.get_platform().split("-")
+        print("Host triplet:", host_triplet)
+
         with open(os.path.join(BLIS_DIR, "make", "%s.jsonl" % platform_arch)) as file_:
             env = {}
             for line in file_:
@@ -228,6 +236,8 @@
                 spec["flags"] = [
                     f for f in spec["flags"] if "visibility=hidden" not in f
                 ]
+                if len(host_triplet) == 4 and host_triplet[0] == "ios":
+                    spec["flags"].append(f"-mios-version-min={host_triplet[1]}")
                 # spec["flags"].append(
                 #    "-D__cpuid=__cpuid_function"
                 # )
