diff --git a/pyobjus/_runtime.h b/pyobjus/_runtime.h
--- a/pyobjus/_runtime.h
+++ b/pyobjus/_runtime.h
@@ -1,6 +1,6 @@
 #include <objc/runtime.h>
 #include <objc/message.h>
-#include <ffi/ffi.h>
+#include <ffi.h>
 #include <stdio.h>
 #include <dlfcn.h>
 #include <string.h>
diff --git a/pyobjus/common.pxi b/pyobjus/common.pxi
--- a/pyobjus/common.pxi
+++ b/pyobjus/common.pxi
@@ -110,7 +110,7 @@
     objc_method_description* protocol_copyMethodDescriptionList(Protocol *p, BOOL isRequiredMethod, BOOL isInstanceMethod, unsigned int *outCount)


-cdef extern from "ffi/ffi.h":
+cdef extern from "ffi.h":
     ctypedef unsigned long  ffi_arg
     ctypedef signed long    ffi_sarg
     ctypedef enum: FFI_TYPE_STRUCT
diff --git a/pyobjus/pyobjus_conversions.pxi b/pyobjus/pyobjus_conversions.pxi
--- a/pyobjus/pyobjus_conversions.pxi
+++ b/pyobjus/pyobjus_conversions.pxi
@@ -428,7 +428,7 @@
         return arg
     elif arg in (True, False):
         return autoclass('NSNumber').alloc().initWithBool_(int(arg))
-    elif isinstance(arg, (str, unicode)):
+    elif isinstance(arg, str):
         return autoclass('NSString').alloc().initWithUTF8String_(arg)
     elif isinstance(arg, int):
         return autoclass('NSNumber').alloc().initWithLong_(arg)
