index defe52c..829037b 100755
--- a/config.sub
+++ b/config.sub
@@ -146,6 +146,7 @@ case $1 in
                        | uclinux-gnu* | kfreebsd*-gnu* | knetbsd*-gnu* | netbsd*-gnu* \
                        | netbsd*-eabi* | kopensolaris*-gnu* | cloudabi*-eabi* \
                        | storm-chaos* | os2-emx* | rtmk-nova* | managarm-* \
+                       | ios*-simulator | tvos*-simulator | watchos*-simulator \
                        | windows-* )
                                basic_machine=$field1
                                basic_os=$maybe_os
@@ -1500,6 +1501,8 @@ case $os in
                obj=$os
                os=
                ;;
+       ios | ios-simulator)
+               ;;
        *)
                # No normalization, but not necessarily accepted, that comes below.
                ;;
@@ -1855,6 +1858,8 @@ case $kernel-$os-$obj in
                ;;
        *-eabi*- | *-gnueabi*-)
                ;;
+       ios-simulator-)
+               ;;
        none--*)
                # None (no kernel, i.e. freestanding / bare metal),
                # can be paired with an machine code file format