diff --git a/pyproj/__init__.py b/pyproj/__init__.py
index 324e34f..a9f14ac 100644
--- a/pyproj/__init__.py
+++ b/pyproj/__init__.py
@@ -90,6 +90,12 @@ __all__ = [
 __proj_version__ = PROJ_VERSION_STR
 proj_version_str = PROJ_VERSION_STR  # pylint: disable=invalid-name
 __proj_compiled_version__ = PROJ_COMPILED_VERSION_STR
+
+# Flet
+import os
+from os.path import dirname
+os.environ["PROJ_DATA"] = f"{dirname(dirname(__file__))}/opt/share/proj"
+
 try:
     pyproj.network.set_ca_bundle_path()
 except DataDirError as err:
diff --git a/setup.py b/setup.py
index 9987cff..b56a0fc 100644
--- a/setup.py
+++ b/setup.py
@@ -194,7 +194,9 @@ def get_extension_modules():
         "include_dirs": include_dirs,
         "library_dirs": library_dirs,
         "runtime_library_dirs": (
-            library_dirs if os.name != "nt" and sys.platform != "cygwin" else None
+            library_dirs
+            if os.name != "nt" and sys.platform != "cygwin" and sys.platform != "ios"
+            else None
         ),
         "libraries": get_libraries(library_dirs),
     }
