diff --git a/setup.py b/setup.py
index ac401dd..a5a9938 100644
--- a/setup.py
+++ b/setup.py
@@ -341,9 +341,7 @@ class pil_build_ext(build_ext):
         return True if value in configuration.get(option, []) else None
 
     def initialize_options(self):
-        self.disable_platform_guessing = self.check_configuration(
-            "platform-guessing", "disable"
-        )
+        self.disable_platform_guessing = True
         self.add_imaging_libs = ""
         build_ext.initialize_options(self)
         for x in self.feature:
@@ -421,10 +419,19 @@ class pil_build_ext(build_ext):
                 self.extensions.remove(extension)
                 break
 
-    def get_macos_sdk_path(self):
+    def get_apple_sdk_path(self):
         try:
+            sdk = {
+                ("ios", False): ["--sdk", "iphoneos"],
+                ("ios", True): ["--sdk", "iphonesimulator"],
+                ("tvs", False): ["--sdk", "appletvos"],
+                ("tvs", True): ["--sdk", "appletvsimulator"],
+                ("watchos", False): ["--sdk", "watchos"],
+                ("watchos", True): ["--sdk", "watchsimulator"],
+                ("darwin", False): [],
+            }[sys.platform, getattr(sys.implementation, "_simulator", False)]
             sdk_path = (
-                subprocess.check_output(["xcrun", "--show-sdk-path"])
+                subprocess.check_output(["xcrun", "--show-sdk-path"] + sdk)
                 .strip()
                 .decode("latin1")
             )
@@ -577,11 +584,15 @@ class pil_build_ext(build_ext):
                 _add_directory(library_dirs, "/usr/X11/lib")
                 _add_directory(include_dirs, "/usr/X11/include")
 
-            sdk_path = self.get_macos_sdk_path()
+            sdk_path = self.get_apple_sdk_path()
+            if sdk_path:
+                _add_directory(library_dirs, os.path.join(sdk_path, "usr", "lib"))
+                _add_directory(include_dirs, os.path.join(sdk_path, "usr", "include"))
+        elif sys.platform in ("ios", "tvos", "watchos"):
+            sdk_path = self.get_apple_sdk_path()
             if sdk_path:
                 _add_directory(library_dirs, os.path.join(sdk_path, "usr", "lib"))
                 _add_directory(include_dirs, os.path.join(sdk_path, "usr", "include"))
-
                 for extension in self.extensions:
                     extension.extra_compile_args = ["-Wno-nullability-completeness"]
         elif (
