diff --git a/src/makefile b/src/makefile
index 42e70bb..d2759e5 100644
--- a/src/makefile
+++ b/src/makefile
@@ -6,34 +6,35 @@ CFLAGS?=-march=native -Wall -O2 -g -fstack-protector-strong -D_FORTIFY_SOURCE=2
         -Warray-bounds -fsanitize=bounds -fsanitize-undefined-trap-on-error -ftrapv $(DEFINES)
         #-fstrict-flex-arrays
 CFLAGS+= -std=c99 -fpic
-LDFLAGS=-g $(LIBS)
+LDFLAGS+=-g $(LIBS)
 CC?=gcc
 AEXT=a
 SOVER=0
+SOEXT=so
 
 AR?=ar
 
-UNAME := $(shell uname -s)
-ARCH := $(shell uname -m)
-ifeq ($(UNAME),Darwin)
-	SOEXT=dylib
-	SOFLAGS=-Wl,-install_name,$(DESTDIR)$(PREFIX)/lib/libopaque.$(SOEXT)
-else
-	CFLAGS+=-Wl,-z,defs -Wl,-z,relro -Wl,-z,noexecstack -Wl,-z,now \
-			  -fsanitize=signed-integer-overflow -fsanitize-undefined-trap-on-error
-	#		  -mbranch-protection=standard -fstrict-flex-arrays=3 
-	SOEXT=so
-	SOFLAGS=-Wl,-soname,libopaque.$(SOEXT).$(SOVER)
-   ifeq ($(ARCH),x86_64)
-      CFLAGS+=-fcf-protection=full
-   endif
-
-   ifeq ($(ARCH),parisc64)
-   else ifeq ($(ARCH),parisc64)
-   else
-      CFLAGS+=-fstack-clash-protection
-   endif
-endif
+# UNAME := $(shell uname -s)
+# ARCH := $(shell uname -m)
+# ifeq ($(UNAME),Darwin)
+# 	SOEXT=dylib
+# 	SOFLAGS=-Wl,-install_name,$(DESTDIR)$(PREFIX)/lib/libopaque.$(SOEXT)
+# else
+# 	CFLAGS+=-Wl,-z,defs -Wl,-z,relro -Wl,-z,noexecstack -Wl,-z,now \
+# 			  -fsanitize=signed-integer-overflow -fsanitize-undefined-trap-on-error
+# 	#		  -mbranch-protection=standard -fstrict-flex-arrays=3 
+# 	SOEXT=so
+# 	SOFLAGS=-Wl,-soname,libopaque.$(SOEXT).$(SOVER)
+#    ifeq ($(ARCH),x86_64)
+#       CFLAGS+=-fcf-protection=full
+#    endif
+
+#    ifeq ($(ARCH),parisc64)
+#    else ifeq ($(ARCH),parisc64)
+#    else
+#       CFLAGS+=-fstack-clash-protection
+#    endif
+# endif
 
 SODIUM_NEWER_THAN_1_0_18 := $(shell pkgconf --atleast-version=1.0.19 libsodium; echo $$?)
 ifeq ($(SODIUM_NEWER_THAN_1_0_18),1)
@@ -58,7 +59,7 @@ ifneq (, $(shell which pandoc))
 endif
 
 
-all: libopaque.$(SOEXT) libopaque.$(AEXT) tests utils/opaque $(MANPAGES)
+all: libopaque.$(SOEXT) libopaque.$(AEXT)
 
 debug: DEFINES=-DTRACE -DNORANDOM
 debug: all
@@ -154,8 +155,7 @@ man-uninstall:
 
 $(DESTDIR)$(PREFIX)/lib/libopaque.$(SOEXT): libopaque.$(SOEXT)
 	mkdir -p $(DESTDIR)$(PREFIX)/lib
-	cp $< $@.$(SOVER)
-	ln -fs $@.$(SOVER) $@
+	cp $< $@
 
 $(DESTDIR)$(PREFIX)/lib/libopaque.$(AEXT): libopaque.$(AEXT)
 	mkdir -p $(DESTDIR)$(PREFIX)/lib
